tools/unit/src/main/kotlin/com/datadog/tools/unit/assertj/JsonElementAssert.kt

```diff
@@ -0,0 +1,57 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.tools.unit.assertj
+
+import com.google.gson.JsonElement
+import java.math.BigDecimal
+import java.math.BigInteger
+import java.util.Date
+import org.assertj.core.api.AbstractObjectAssert
+import org.assertj.core.api.Assertions.assertThat
+
+/**
+ * Assertion methods for [JsonElement].
+ */
+class JsonElementAssert(actual: JsonElement) :
+    AbstractObjectAssert<JsonElementAssert, JsonElement>(actual, JsonElementAssert::class.java) {
+
+    /**
+     *  Verifies that the actual jsonElement matches the expected value.
+     *  @param expected the expected
+     */
+    fun isJsonValueOf(expected: Any?): JsonElementAssert {
+        when (expected) {
+            null -> assertThat(actual.isJsonNull()).isTrue()
+            is Boolean -> assertThat(actual.asBoolean).isEqualTo(expected)
+            is Int -> assertThat(actual.asInt).isEqualTo(expected)
+            is Long -> assertThat(actual.asLong).isEqualTo(expected)
+            is Float -> assertThat(actual.asFloat).isEqualTo(expected)
+            is Double -> assertThat(actual.asDouble).isEqualTo(expected)
+            is String -> assertThat(actual.asString).isEqualTo(expected)
+            is Date -> assertThat(actual.asString).isEqualTo(expected)
+            is BigInteger -> assertThat(actual.asBigInteger).isEqualTo(expected)
+            is BigDecimal -> assertThat(actual.asBigDecimal).isEqualTo(expected)
+            is List<*> -> JsonArrayAssert(actual.asJsonArray).isJsonArrayOf(expected)
```

Comment:
should we support `Map` here as well?

Reply:
For now I don't think we need to as it's unused, we can still add it later if the use case presents itself